easybuild/easyconfigs/g/GATK/GATK-4.0.10.0-foss-2018b-Python-3.6.6.eb

```diff
@@ -0,0 +1,55 @@
+##
+# EasyBuild Easyconfig
+#
+# Copyright:: Copyright 2012-2013 Cyprus Institute /
+# CaSToRC, University of Luxembourg / LCSB
+# Authors::   George Tsouloupas <g.tsouloupas@cyi.ac.cy>,
+#             Fotis Georgatos <fotis.georgatos@uni.lu>,
+#             Kenneth Hoste (UGent)
+# License::   MIT/GPL
+# $Id$
+#
+# This work implements a part of the HPCBIOS project and is a component
+# of the policy: http://hpcbios.readthedocs.org/en/latest/HPCBIOS_2012-94.html
+# Modified by: Adam Huffman, Jonas Demeulemeester
+# The Francis Crick Institute
+# Modified for version 4.0.5.1 by: Ruben van Dijk, University of Groningen
+##
+
+easyblock = 'Tarball'
+
+name = 'GATK'
+version = '4.0.10.0'
+versionsuffix = '-Python-%(pyver)s'
+pyver = '3.6.6'
+javaver = '1.8.0_181'
+
+homepage = 'http://www.broadinstitute.org/gatk/'
+description = """The Genome Analysis Toolkit or GATK is a software package
+developed at the Broad Institute to analyse next-generation resequencing
+data. The toolkit offers a wide variety of tools, with a primary focus on
+variant discovery and genotyping as well as strong emphasis on data quality
+assurance. Its robust architecture, powerful processing engine and
+high-performance computing features make it capable of taking on projects
+of any size."""
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+source_urls = ['https://github.com/broadinstitute/gatk/releases/download/%(version)s/']
+sources = ['gatk-%(version)s.zip']
+checksums = ['6a149750dfcdb250e07198cf1726d2f0958d2b8f257618b3b585194c4c392283']
+
+dependencies = [
+    ('Python', pyver),
+    ('Java', javaver, '', True),
```

Comment:
@fizwit You don't need to define `pyver` and `javaver` as local variables in the easyconfig file. The `%(pyver)s` and `%(javaver)s` templates are defined automatically by EasyBuild when `Python` & `Java` are included in `dependencies`.

Also, for `Java` we have started using a wrapper rather than an explicit version, which allows in-place updating; see https://easybuild.readthedocs.io/en/latest/Wrapping_dependencies.html for more details.

With that in mind, please update this to:

```python
dependencies = [
    ('Python', '3.6.6'),
    ('Java', '1.8', '', True),
]
```

and remove the `pyver = ...` and `javaver = ...` lines above

Reply:
dependencies = [
    ('Python', '3.6.6'),
    ('Java', '1.8', '', True),
]
My build environment is an Ubuntu container which has EasyBuild installed and 2018b. The container is as minimal as possible, ie no local Java.  github/FredHutch/ls2_easybuild_toolchain

With easybuild 3.7.0 the above dependency definition gives me this error. 

== FAILED: Installation ended unsuccessfully (build directory: /app/build/GATK/4.0.10.0/foss-2018b-Python-3.6.6): build failed (first 300 chars): Missing modules for one or more dependencies: Java/1.8
== Results of the build can be found in the log file(s) /tmp/eb-N_DTog/easybuild-GATK-4.0.10.0-20181009.134455.YPcdE.log
ERROR: Build of /home/neo/GATK-4.0.10.0-foss-2018b-Python-3.6.6.eb failed (err: 'build failed (first 300 chars): Missing modules for one or more dependencies: Java/1.8')

Reply:
@fizwit You should be able to fix that by simply using `--robot`, which will install `Java/1.8` (a wrapper for `Java/1.8.0_181`) first?

Reply:
our build environment does not have TCL defined.  We do not build Modules with TCL support.  TCL seems to be a requirement?

eb GATK-4.0.10.0-foss-2018b-Python-3.6.6.eb --robot
== temporary log file in case of crash /tmp/eb-rndHtv/easybuild-fuZcyD.log
== resolving dependencies ...
== processing EasyBuild easyconfig /app/software/EasyBuild/3.7.0/lib/python2.7/site-packages/easybuild_easyconfigs-3.7.0-py2.7.egg/easybuild/easyconfigs/j/Java/Java-1.8.eb
== building and installing Java/1.8...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== taking care of extensions...
== postprocessing...
== sanity checking...
== FAILED: Installation ended unsuccessfully (build directory: /app/build/Java/1.8/dummy-dummy): build failed (first 300 chars): Module command 'module use --priority 10000 /tmp/eb-rndHtv/tmp0A57Jn/all/' failed with exit code 1; stderr: sh: 1: UNKNOWN_TCLSH: not found
Lmod has detected the following error: Unable to parse:
"/tmp/eb-rndHtv/tmp0A57Jn/all/Java/.modulerc". Aborting!



; stdout:
false

== Results of the build can be found in the log file(s) /tmp/eb-rndHtv/easybuild-Java-1.8-20181009.184414.rDVOe.log
ERROR: Build of /app/software/EasyBuild/3.7.0/lib/python2.7/site-packages/easybuild_easyconfigs-3.7.0-py2.7.egg/easybuild/easyconfigs/j/Java/Java-1.8.eb failed (err:'build failed (first 300 chars): Module command \'module use --priority 10000 /tmp/eb-rndHtv/tmp0A57Jn/all/\' failed with exit code 1; stderr: sh: 1: UNKNOWN_TCLSH: not found\nLmod has detected the following error: Unable to parse:\n"/tmp/eb-rndHtv/tmp0A57Jn/all/Java/.modulerc". Aborting!\n\n\n\n; stdout: \nfalse\n')

cat

Reply:
https://github.com/easybuilders/easybuild-framework/blob/master/easybuild/tools/module_generator.py
line 203

Reply:
This was fixed in https://github.com/easybuilders/easybuild-framework/pull/2597 (to be included with EasyBuild v3.7.1), but it does require Lmod 7.8 or newer (since that's what's needed for Lmod to support picking up `.modulerc.lua`).